Adding N2WS Backup and Recovery integration (#748)
DataDog/integrations-extras: n2ws/README.md

[external_monitoring]
enabled=True
run ‘service apache2 restart’

4.	Install Datadog Agent on N2WS Instance
Login to Datadog and go to Integrations -> Agent -> Ubuntu
Copy the agent ‘easy one-step install’ command 
Connect to your N2WS Backup and Recovery Instance with SSH Client, type “sudo su” and  run the agent Install command.

5.	Setup Datadog Dashboard metrics
Login to Datadog and go to ‘Metrics-> Explorer’

From Graph: 
Select your metric from the list (all N2WS metrics begin with the string ‘cpm_<metric-name>’)

From Over:
Select data from the list (all N2WS users data begin with the string ‘cpm:user:<user-name>’)
You can either select a specific user or the entire N2WS instance alternatively

6.	Configure Datadog Dashboard, and select the desired data to monitor
Create new Datadog dashboard. 
Add a Timeseries graph or any other desired graph (by dragging and dropping it to the dashboard)
Edit the desire data to be monitored in your graph
Save the graph settings
 
Users can alternatively load a ready-made datadog template to monitor their CPM with datadog client
https://support.n2ws.com/portal/en/kb/articles/datadog-templates

> 

Reviewer: I see this PR includes some dashboards. Once this PR is merged there will be a tile in app for N2WS integration too. If users click install on it, then the dashboards will get installed on their account.

# N2WS Backup & Recovery

## Overview

N2WS Backup & Recovery (CPM), known as N2WS, is an enterprise-class backup, 
recovery and disaster recovery solution for the Amazon Web Services (AWS). 
Designed from the ground up to support AWS, N2WS uses cloud native technologies 
(e.g. EBS snapshots) to provide unmatched backup and, more importantly, restore capabilities in AWS.

N2WS Backup and Recovery Instance supports the monitoring of backups, DR, copy to S3, alerts 
and more by Datadog monitoring service. 
It will allow users to monitor and analyse the N2WS Backup and Recovery Dashboard metrics.

## Setup

### Installation

Please perform the following steps in order to activate the service and to monitor your N2WS instance:

1.	Setup Datadog Account 
Visit Datadog to: https://www.datadoghq.com/pricing/ and setup an account that fits your scale.

2.	Install Python Integration
Login to Datadog and go to Integrations -> Integrations

> Search for ‘Python’ and install it

Author: Does this link valid for all DataDog accounts regardless of regional location?

3.	Enable Datadog support on N2WS Instance
Connect to your N2WS Backup and Recovery Instance with SSH Client, type “sudo su” and add the following lines to /cpmdata/conf/cpmserver.cfg
[external_monitoring]
enabled=True

> run ‘service apache2 restart’

Author: Using back-ticks `` , right?

4.	Install Datadog Agent on N2WS Instance
Login to Datadog and go to Integrations -> Agent -> Ubuntu
Copy the agent ‘easy one-step install’ command 

> Connect to your N2WS Backup and Recovery Instance with SSH Client, type “sudo su” and  run the agent Install command.

Reviewer 2: Instead of walking through that whole thing, let's just say:

Connect to your N2WS Backup and Recovery instance with SSH and run the following command to install the Datadog Agent. You may require `sudo` privileges to perform this action: `<insert command here>`

Author: Every DataDog account has a different command, that is why we can't insert a command here.

5.	Setup Datadog Dashboard metrics

> Login to Datadog and go to ‘Metrics-> Explorer’

Reviewer 2: Rather than navigating through, let's just link to the Metrics Explorer - https://app.datadoghq.com/metric/explorer

> From Graph: 
> Select your metric from the list (all N2WS metrics begin with the string ‘cpm_metric’)
> 
> From Over:
> Select data from the list (all N2WS users data begin with the string ‘cpm:user:<user-name>’)
> You can either select a specific user or the entire N2WS instance alternatively

Reviewer 2: ```suggestion
**Graph**: Select your metric from the list. All N2WS metrics begin with the string ‘cpm_metric’.

**Over**: Select data from the list. All N2WS user data begins with the string ‘cpm:user:<user-name>’. You can select either a specific user or the entire N2WS instance.
```

6.	Get N2WS dashboards

> In Datadog's UI, go to 'Integrations-> Integrations', search for 'N2WS' tile and install it. You will get 3 types of dashboards to your account:

Reviewer 2: Just link to the integration rather than giving directions.

Author: I guess that the link will look like that: https://app.datadoghq.com/account/settings#integrations/N2WS

If it will look different, please tell us

Author: In addition, I think that the user is already in integration tile when he reads this text, isn't he?

'N2WSBackup&Recovery-Graphicalversion', 'N2WSBackup&Recovery-Graphicalversion-areas' and 'N2WSBackup&Recovery-Squaresdashboard'.
 

> Users can alternatively import the .json templates for N2WS from here:
> https://support.n2ws.com/portal/en/kb/articles/datadog-templates

Reviewer 2: ```suggestion
Alternatively, users can [import JSON templates from N2WS][link here].
```

put links at bottom of this file (see other integrations readmes for how we typically do this)

Author: Will it become a clickable text?

Reviewer: yes

## Data Collected

> Datadog collects data about N2WS Backup & Recovery backups: Number of snapshots of each types, successfull backups, failed backups, partially successful backups, 

Reviewer 2: ```suggestion
Datadog collects data about N2WS Backup & Recovery backups: number of snapshots of each type, successful backups, failed backups, partially successful backups, 
```

> protected resources from any type, data about volumes capacity, alerts and etc.

Reviewer 2: ```suggestion
protected resources from any type, data about volume capacity, alerts, etc.
```

### Metrics

> See [metadata.csv] for a list of metrics provided by this check.

Reviewer 2: Missing the link here.

### Events

Datadog collects alert messages from all the N2WS Backup & Recovery hosts.

### Service Checks

N2WS Backup & Recovery doesn't have service checks.

## Troubleshooting

Please take a look at the User Guide and documentation: https://n2ws.com/support/documentation

> Support team is available for you at: https://n2ws.com/support 

Reviewer 2: Let's remove this section. We assume that users are N2WS users and can access the documentation and support team themselves. Unless there is something of note that integration users should specifically be aware of in the N2WS docs, we don't need to link them.

Author: Since we are considering getting to DataDog's marketplace in the future, the users might be DataDog's customers that will need this link.
I would keep it as is.
